frontend/src/features/Settings/Settings.tsx

```diff
@@ -0,0 +1,205 @@
+
+    return (
+      <form
+        className="grid gap-6 bg-gray-800 p-12 w-full max-w-4xl mx-auto rounded-lg shadow-lg"
+        onSubmit={(e) => e.preventDefault()}
+        style={{
+          height: "80vh",
+          display: "flex",
+          flexDirection: "column",
+          justifyContent: "center",
+        }}
+      >
+        <h1 className="text-3xl font-extrabold text-center mb-6">
+          User Settings
+        </h1>
+        {/* User Name */}
+        <div>
+          <label className="block font-bold text-gray-400 mb-2">
+            User Name
+          </label>
+          <input
+            type="text"
+            className="w-full p-3 rounded bg-gray-700 text-white focus:outline-none focus:ring-2 focus:ring-blue-500"
+            value={settings.userName}
+            onChange={(e) => handleInputChange("userName", e.target.value)}
+          />
+          {/* TODO: Don't allow viewing the token, only allow changing it */}
+          {/* Token */}
+          <label className="block font-bold text-gray-400 mt-4 mb-2">
+            Token Input
+          </label>
+          <input
+            type="text"
+            className="w-full p-3 rounded bg-gray-700 text-white focus:outline-none focus:ring-2 focus:ring-blue-500"
+            value={settings.token}
+            onChange={(e) => handleInputChange("token", e.target.value)}
+          />
+        </div>
+        {/* Preferences */}
+        <div>
+          <label className="block font-bold text-gray-400 mb-2">
+            Preferences
+          </label>
+          <div className="flex gap-4 items-center">
+            <label className="flex items-center gap-2">
+              <input
+                type="checkbox"
+                className="form-checkbox"
+                checked={settings.preferences.darkMode}
+                onChange={(e) =>
+                  handlePreferenceChange("darkMode", e.target.checked)
+                }
+              />
+              <span className="text-white">Dark Mode</span>
+            </label>
+            <div>
+              <label className="block text-gray-400 mb-1">Default Scale</label>
+              <input
+                type="number"
+                className="w-24 p-2 rounded bg-gray-700 text-white focus:outline-none focus:ring-2 focus:ring-blue-500"
+                value={settings.preferences.defaultScale}
+                onChange={(e) =>
+                  handlePreferenceChange("defaultScale", Number(e.target.value))
+                }
+              />
+            </div>
+          </div>
+        </div>
+        {/* Save Button */}
+        <button
+          className="transition-all ease-in-out duration-300 bg-green-600 text-white font-bold rounded-lg py-2 px-4
+                     hover:bg-green-700 hover:scale-105 focus:outline-none focus:ring-2 focus:ring-green-500"
+          type="button"
+          onClick={void handleSave}
```

Comment:
Ahhh, yup! It was working until I went to push and eslint complained about this very thing

Reply:
I figured that was the case haha it's sort of a weird thing to try to figure out since the warning isn't exactly the most clear. But hopefully that makes sense!